### DISCUSSION TITLE
Callback from charts

### DISCUSSION THREAD

**Discussion by armeya25** [highfestiva/finplot#438] [Q&A] [answered]
Is there any option for call back function in finplot.

Eg. I want to get value points on which line is drawn,
Is it possible to get it's value while drawing line on charrt

**Comment by highfestiva** [comment]
There are, possibly. I don't understand exactly what you're looking for though. finplot has callbacks for mouse-over, click, double-click, timer and for modifying crosshair text. And it's trivial to re-route the keyboard function for your own purposes.

You probably don't mean "while drawing," as the drawing of most things takes milliseconds, and there is no way you're able to read that fast. I'm guessing you mean hover/mouse over, so have a look here: https://github.com/highfestiva/finplot/blob/master/finplot/examples/snp500.py#L57

**Reply by armeya25** [reply]
I mean I draw say a interactive line and dragged it elsewhere, I wish to get those co-ordinates where it is now to in my terminal or say to variable,

Those like implemented  in Dearpygui library's callback

**Reply by highfestiva** [reply, marked as answer]
Good idea, I added a function which you can use to override with your own line method, so you can get information of when it's moved and stuff like that. Patch your finplot installation with the commit 693744d. The function is called `_create_poly_line`. Implement something like this:

```python
class MyPolyLine:
    def __init__(self, vb, *args, **kwargs):
        super().__init__(vb, *args, **kwargs)

finplot._create_poly_line = MyPolyLine
```

Override the relevant methods. See [here](https://github.com/highfestiva/finplot/blob/master/finplot/__init__.py#L690) for what methods currently exist. GL!
